### PR TITLE
added a maintenance DAG for nci database

### PR DESCRIPTION
**Pull request by NikitaGandhi**
executing `vacuum analyze` to nci db every second day. this will optimise the database and improve performance for nci explorer

### CODE REVIEW

File: dags/k8s_nci_db_maintenance.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    "-c",
+    dedent(
+        """
+            # agdc tables
+            psql -h $(DB_HOSTNAME) -U $(DB_ADMIN_USER) -d $(DB_DATABASE) -c \
+            "vacuum verbose analyze agdc.dataset, agdc.dataset_type, agdc.dataset_source, agdc.metadata_type, agdc.dataset_location;"
+            
+            # cubedash tables
+            psql -h $(DB_HOSTNAME) -U $(DB_ADMIN_USER) -d $(DB_DATABASE) -c \ 
+            "cubedash.dataset_spatial, cubedash.product, cubedash.region, cubedash.time_overview;"
```
Comment (jeremyh) [inline dags/k8s_nci_db_maintenance.py:81]: Missed the prefix in the second command

```suggestion
            "vacuum verbose analyze cubedash.dataset_spatial, cubedash.product, cubedash.region, cubedash.time_overview;"
```

**Review by jeremyh — COMMENTED** [review]
Thanks for adding this :)

File: dags/k8s_nci_db_maintenance.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    default_args=DEFAULT_ARGS,
+    catchup=False,
+    concurrency=1,
+    max_active_runs=1,
+    tags=["k8s"],
+    schedule_interval="10 2 * * 2,4,6",  # every second day 2:10AM
+    dagrun_timeout=timedelta(minutes=60 * 4),
+)
+
+affinity = {
```
Comment (pindge) [inline dags/k8s_nci_db_maintenance.py:50]: can u import affinity from infra/podconfig.py

**Review by pindge — APPROVED** [review]

File: dags/k8s_nci_db_incremental_update_summary.py [hunk 3]
```diff
@@ -56,21 +57,7 @@
...
-                    "operator": "In",
-                    "values": [
-                        "ondemand",
-                    ]
-                }]
-            }]
-        }
-    }
-}
+affinity = NODE_AFFINITY
```
Comment (whatnick) [inline dags/k8s_nci_db_incremental_update_summary.py:60]: Could we call this ONDEMAND_NODE_AFFINITY

**Pull request merged**